web/src/engine/websites/CopyManga.ts

```diff
@@ -0,0 +1,161 @@
+import { Tags } from '../Tags';
+import icon from './CopyManga.webp';
+import { Chapter, DecoratableMangaScraper, Manga, Page, type MangaPlugin } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { Check, Choice, Text } from '../SettingsManager';
+import { EngineResourceKey as E, WebsiteResourceKey as W } from '../../i18n/ILocale';
+import { FetchRequest } from '../FetchProviderNodeWebkit';
+import { FetchJSON } from '../FetchProvider';
+
+type APIResponse<T> = {
+    code: number,
+    message: string,
+    results: T
+}
+
+type APISingleComic = {
+    comic: APIComic
+}
+
+type APIResultList<T> = {
+    list: T[]
+}
+
+type APIPages = {
+    chapter: APIChapter
+}
+
+type APIComic = {
+    uuid: string,
+    path_word: string,
+    name: string
+}
+
+type APIChapter = {
+    uuid: string,
+    name: string,
+    contents: {
+        url: string
+    }[];
+    words: number[]
+}
+
+@Common.ImageAjax()
+export default class extends DecoratableMangaScraper {
+
+    private readonly API_HEADERS = {
+        'User-Agent': '"User-Agent" to "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/102.0.5005.124 Safari/537.36 Edg/102.0.1245.44"',
+        version: new Date().toISOString().replace(/-/g, ".").split('T').shift(),
+        platform: "1",
+    };
+
+    private readonly apiurl = 'https://api.copymanga.org';
+
+    public constructor() {
+        super('copymanga', 'CopyManga', 'https://www.copymanga.site', Tags.Media.Manhwa, Tags.Media.Manhua, Tags.Media.Manga, Tags.Language.Chinese, Tags.Source.Aggregator);
+
+        //URL
+        this.Settings.url = new Text('urloverride', W.Plugin_Settings_UrlOverride, W.Plugin_Settings_UrlOverrideInfo, 'https://www.copymanga.site');
```

Comment:
May simplify to use domain instead of URL, as almost all websites are using `https` and are running on port `80`

Reply:
Consider using `this.URI.href` or `this.URI.hostname` as default value to avoid duplicated strings